common/flatpak-dir.c

```diff
@@ -14607,19 +14607,14 @@ flatpak_dir_get_default_locale_languages (FlatpakDir *self)
   int i;
 
   extra_languages = flatpak_dir_get_config_strv (self, "xa.extra-languages");
-  for (i = 0; extra_languages != NULL && extra_languages[i] != NULL; i++)
-    {
-       gchar **locales = g_strsplit (extra_languages[i], "_", -1);
-       if (g_strv_length (locales) > 1)
-         extra_languages[i] = locales[0];
-       g_free(locales);
-    }
+  for (i = 0; extra_languages != NULL && extra_languages[i] != NULL && strlen (extra_languages[i]) > 3; i++)
+    extra_languages[i] = g_strsplit (extra_languages[i], "_", -1)[0];
```

Comment:
This is worse, memory-management-wise.

flatpak_dir_get_config_strv returns an allocated string array that needs to be freed with g_strfreev (or implicitly, by declaring it as g_auto(GStrv)). So does g_strsplit(). If you want to do this by poking at extra_languages directly, it needs to be

```
  for (i = 0...)
    {
       g_auto(GStrv) s = g_strsplit (extra_languages[i], "_", -1);
       g_free (extra_languages[i]);
       extra_languages[i] = g_strdup (s[0]);
    }
```

Reply:
Oh! got it. My previous attempt was something like this, but I didn't know you have to g_strdup() to pass the data, so that wasn't quite working..